File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    var answer = [];
    
    answer.push(arr[0]);
    let j = 0;
    for(let i = 1; i < arr.length; i++){
        if(arr[i] !== answer[j]){
            answer.push(arr[i]);
            j++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        
        ArrayList<Integer> array = new ArrayList<>();
        
        for(int n : arr) {
            if(array.size() == 0) {
                array.add(n);
            }
            else {
                if(array.get(array.size() - 1) != n) {
                    array.add(n);
                }
            }
        }

        return array.stream().mapToInt(Integer::intValue).toArray();
    }
}